File: frontend/src/domain/entities/Site.ts
export interface Site {
  id: string;
  name: string;
  location: string;
  description: string;
  latitude?: number;
  longitude?: number;
  oltCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface CreateSiteDto {
  name: string;
  location?: string;
  description?: string;
  latitude?: number;
  longitude?: number;
}

export interface UpdateSiteDto {
  name?: string;
  location?: string;
  description?: string;
  latitude?: number;
  longitude?: number;
}

File: backend/internal/models/site.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Site struct {
	ID          uuid.UUID `gorm:"type:uuid;primaryKey"`
	Name        string    `gorm:"type:varchar(255);not null"`
	Location    string    `gorm:"type:text"`
	Description string    `gorm:"type:text"`
	Latitude    *float64  `gorm:"type:double precision" json:"latitude,omitempty"`
	Longitude   *float64  `gorm:"type:double precision" json:"longitude,omitempty"`
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

func (s *Site) BeforeCreate(tx *gorm.DB) error {
	if s.ID == uuid.Nil {
		s.ID = uuid.New()
	}
	return nil
}

func (s *Site) TableName() string {
	return "sites"
}
